plan/physical_plan_builder.go

```diff
@@ -677,33 +735,72 @@ func (p *Union) convert2PhysicalPlan(prop *requiredProperty) (*physicalPlanInfo,
 		childInfos = append(childInfos, info)
 	}
 	info = p.matchProperty(prop, childInfos...)
+	info = enforceProperty(limitProperty(limit), info)
```

Comment:
We should call this function upon every operator that refuses the required property.